### DISCUSSION TITLE
issues with .authWithOAuth2

### DISCUSSION THREAD

**Discussion by ainmon** [pocketbase/pocketbase#5287] [Q&A] [answered]
Hi all, I am attempting to setup OAuth2 login for a simple sveltekit/pocketbase application. I have been able to do this in the past using more of the manual approach but I noticed the new functionality and am trying to work with it.

This is in my svelte component and I've verified the google redirects are set to /api/oauth2-redirect. I'm not sure if I'm misunderstanding something? Any help would be appreciated!

```
const startOAuth2Flow = async (providerName) => {
        try {
            const authData = await pb.collection('users').authWithOAuth2({ provider: providerName });
        } catch (error) {
            console.error('Error starting OAuth2 flow:', error);
        }
    };
```

**Comment by ganigeorgiev** [comment]
It is not clear from your description what actually is the issue you are experiencing, nor the execution context of your app.

Is there any error thrown?

How do you even communicate with PocketBase from your SvelteKit app - only client-side, only server-side, both? 

The "all in one" OAuth2 flow works only client-side because it needs to be able to start a realtime connection to receive the success result and open a browser window with the provider's login page. Under the hood, `authWithOAuth2` calls `authWithOAuth2Code` (aka. the "manual" flow). It is just a convenient method for the cases where a custom redirect/deep-link is not always easily done (SPAs, mobile apps, desktop apps, cli, etc.).

_As a side-note, if you are new to web development I really don't recommend using JS SSR with PocketBase. There will be more detailed note regarding this in the documentation in the near future, but for now you can read some rants in https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7620032, https://github.com/pocketbase/pocketbase/discussions/4412#discussioncomment-8575834, https://github.com/pocketbase/pocketbase/discussions/5190#discussioncomment-9999444 and probably others._

**Reply by ainmon** [reply]
I apologize for the confusion; I'm getting 'Error 400: redirect_uri_mismatch' when attempting this. The attempt is to do this from the client-side only. For context, it is simply working off of the homepage and just using a simple button to call this auth method. The error would make it seem like I just failed to setup the correct redirects on google but I verified those are what the documentation says to specify. My understanding is that the redirect url is handled already so I'm not quite sure if I missed something?

**Reply by ganigeorgiev** [reply, marked as answer]
Make sure that the redirect url doesn't have typos or extra characters. Additionally note that if you are testing on localhost note that most provider treat `127.0.0.1` and `localhost` as separate addresses, so the redirect host in your Google app must match with the url you've used to initialize the JS SDK.
Or in other words, if you have `const pb = new PocketBase("http://127.0.0.1:8090")`, then the redirect url must be `http://127.0.0.1:8090/api/oauth2-redirect`.

**Reply by ainmon** [reply]
This was the solution. Thank you! That makes a lot of sense in hindsight but I was thinking it had to come from the domain.
